src/messagepublishers/imupublisher.h

```diff
@@ -59,19 +60,25 @@ static void variance_from_stddev_param(std::string param, double *variance_out)
 
 struct ImuPublisher : public PacketCallback
 {
-    ros::Publisher pub;
+    ros::Publisher inner_pub;
+    diagnostic_updater::Updater updater;
+    double expected_frequency;
+    double frequency_tolerance = 0.1;
+    diagnostic_updater::DiagnosedPublisher<sensor_msgs::Imu> pub;
     std::string frame_id = DEFAULT_FRAME_ID;
 
 
     double orientation_variance[3];
     double linear_acceleration_variance[3];
     double angular_velocity_variance[3];
 
-    ImuPublisher(ros::NodeHandle &node)
+    ImuPublisher(ros::NodeHandle &node) :
+        expected_frequency(ros::param::param("~expected_frequency", 100)),
+        inner_pub(node.advertise<sensor_msgs::Imu>("imu/data", ros::param::param("~publisher_queue_size", 5))),
+        updater(node),
+        pub(inner_pub, updater, {&expected_frequency, &expected_frequency, frequency_tolerance}, {})
     {
-        int pub_queue_size = 5;
-        ros::param::get("~publisher_queue_size", pub_queue_size);
-        pub = node.advertise<sensor_msgs::Imu>("imu/data", pub_queue_size);
+        updater.setHardwareID("none");
```

Comment:
Should this be a parameter? Don't think we use it, but some people filter on diagnostic hardware-ids?

Reply:
Isn't the hardware-id mostly generic, like 'imu', 'lidar', 'gps' ?

Reply:
No, I think it should be a model number/serial number, or `none`

Reply:
Alright!